CONTRIBUTING.md

```diff
@@ -0,0 +1,131 @@
+Contributing to Slicer
+======================
+
+There are many ways to contribute to Slicer, with varying levels of effort.  Do try to
+look through the [documentation](https://www.slicer.org/wiki/Documentation/Nightly/Developers) first if something is unclear, and let us know how we can
+do better.
+
+  * Ask a question on the [slicer-devel email list](http://massmail.spl.harvard.edu/mailman/listinfo/slicer-devel)
+  * Submit a feature request or bug, or add to the discussion on the [Slicer issue tracker](http://na-mic.org/Mantis)
+  * Submit a [Pull Request](https://github.com/Slicer/Slicer/pulls) to improve Slicer or its documentation
+
+We encourage a range of Pull Requests, from patches that include passing tests and
+documentation, all the way down to half-baked ideas that launch discussions.
+
+The PR Process, Circle CI, and Related Gotchas
+----------------------------------------------
+
+#### How to submit a PR ?
+
+If you are new to Slicer development and you don't have push access to the Slicer
+repository, here are the steps:
+
+1. [Fork and clone](https://help.github.com/articles/fork-a-repo/) the repository.
+3. Create a branch.
+4. [Push](https://help.github.com/articles/pushing-to-a-remote/) the branch to your GitHub fork.
+5. Create a [Pull Request](https://github.com/Slicer/Slicer/pulls).
+
+This corresponds to the `Fork & Pull Model` mentioned in the [GitHub flow](https://guides.github.com/introduction/flow/index.html)
+guides.
+
+When submitting a PR, the developers following the project will be notified. That
+said, to engage specific developers, you can add `Cc: @<username>` comment to notify
+them of your awesome contributions.
+Based on the comments posted by the reviewers, you may have to revisit your patches.
+
+
+#### How to efficiently contribute ?
+
+We encourage all developers to:
+
+* review and follow the style guidelines described
+[on the wiki](https://www.slicer.org/wiki/Documentation/Nightly/Developers/Style_Guide#Commits).
+
+* consider potential backward compatibility breakage and discuss these on the
+  mailing list. For example, update of ITK, Python, Qt or VTK version, change to
+  core functionality, should be carefully reviewed and integrated. Ideally, several
+  developers would test that the changes don't break extensions.
+
+#### How to integrate a PR ?
+
+Getting your contributions integrated is relatively straightforward, here
+is the checklist:
+
+* All tests pass
```

Comment:
Should you suggest here that your addition should be tested ? Something along the lines of:  
"Ideally, contributions should include new or improved automatic testing.  For reference, see the testing [How-tos](https://www.slicer.org/wiki/Documentation/Nightly/Developers/Tutorials/Testing)".